src/Parser/Multipart.php

```diff
@@ -0,0 +1,251 @@
+<?php
+
+namespace React\Http\Parser;
+
+use Evenement\EventEmitterTrait;
+use React\Http\File;
+use React\Http\Request;
+use React\Stream\ThroughStream;
+use React\Stream\Util;
+
+class Multipart implements ParserInterface
+{
+    use EventEmitterTrait;
+
+    /**
+     * @var string
+     */
+    protected $buffer = '';
+
+    /**
+     * @var string
+     */
+    protected $ending = '';
+
+    /**
+     * @var int
+     */
+    protected $endingSize = 0;
+
+    /**
+     * @var string
+     */
+    protected $boundary;
+
+    /**
+     * @var Request
+     */
+    protected $request;
+
+
+    public function __construct(Request $request)
+    {
+        $this->request = $request;
+        $headers = $this->request->getHeaders();
+        $headers = array_change_key_case($headers, CASE_LOWER);
+        preg_match('/boundary="?(.*)"?$/', $headers['content-type'], $matches);
+
+        $dataMethod = 'findBoundary';
+        if (isset($matches[1])) {
+            $this->setBoundary($matches[1]);
+            $dataMethod = 'onData';
+        }
+        $this->request->on('data', [$this, $dataMethod]);
+        Util::forwardEvents($this->request, $this, [
+            'end',
```

Comment:
I guess this is wrong. 'end' is called on request when the connection is closed, right? this explains maybe why the deferred parser only continues when the connection closes. Actually 'end' on the parser should only be called when the request body is sent, not when the connection closes.